// @ionic-native/native-page-transitions/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var NativePageTransitions = /** @class */ (function (_super) {
    __extends(NativePageTransitions, _super);
    function NativePageTransitions() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NativePageTransitions.prototype.slide = function (options) { return cordova(this, "slide", {}, arguments); };
    NativePageTransitions.prototype.flip = function (options) { return cordova(this, "flip", {}, arguments); };
    NativePageTransitions.prototype.fade = function (options) { return cordova(this, "fade", { "platforms": ["iOS", "Android"] }, arguments); };
    NativePageTransitions.prototype.drawer = function (options) { return cordova(this, "drawer", { "platforms": ["iOS", "Android"] }, arguments); };
    NativePageTransitions.prototype.curl = function (options) { return cordova(this, "curl", { "platforms": ["iOS"] }, arguments); };
    NativePageTransitions.prototype.executePendingTransition = function () { return cordova(this, "executePendingTransition", {}, arguments); };
    NativePageTransitions.prototype.cancelPendingTransition = function () { return cordova(this, "cancelPendingTransition", {}, arguments); };
    NativePageTransitions.pluginName = "NativePageTransitions";
    NativePageTransitions.plugin = "com.telerik.plugins.nativepagetransitions";
    NativePageTransitions.pluginRef = "plugins.nativepagetransitions";
    NativePageTransitions.repo = "https://github.com/Telerik-Verified-Plugins/NativePageTransitions";
    NativePageTransitions.platforms = ["Android", "iOS", "Windows Phone 8"];
NativePageTransitions.ɵfac = function NativePageTransitions_Factory(t) { return ɵNativePageTransitions_BaseFactory(t || NativePageTransitions); };
NativePageTransitions.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: NativePageTransitions, factory: function (t) { return NativePageTransitions.ɵfac(t); } });
var ɵNativePageTransitions_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(NativePageTransitions);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NativePageTransitions, [{
        type: Injectable
    }], null, null); })();
    return NativePageTransitions;
}(IonicNativePlugin));
export { NativePageTransitions };
